### ISSUE TITLE
Timeframe for the Hosted Agents to have .NET 4.7.2?

### ISSUE THREAD

**Issue by kelps** [Microsoft/vsts-image-generation#141]
We use Hosted Agents for our builds and we are planning on changing our main product to .NET 4.7.2 to take advantage of SameSite cookies and DI in ASPNET (not Core). Always tricky to find information on when new features and frameworks will be available to us on Hosted Agents CI. We really don't want to create and maintain our own build servers.

**Comment by lokesh755** [comment]
We've generated new image yesterday.. Will update all hosted agents by next week.

**Comment by kelps** [comment]
It would be nice to have a document showing the planned timeframes and how the update process work 😉 #141

**Comment by kelps** [comment]
Also: is the document at https://github.com/Microsoft/vsts-image-generation/blob/master/images/win/Vs2017-Server2016-Readme.md updated automagically when the agents are? How much can we trust that this document reflects the current reality?

**Comment by bryanmacfarlane** [comment]
@kelps - we have in the build definition, there is (or will be soon for your account) a details link that will open the markdown in a flyout.  That will auto magically get updated when the image shows up in your scale unit for your account.  The admin UI will also have a details tab showing the same.  The doc you linked about gets updated in GH when we generate the image to start rolling out.  We also branch for each monthly release.  The rolling out is a minimum 4+ days across scale units (we roll and wait for issues, continue).

**Comment by kelps** [comment]
thanks for the info @bryanmacfarlane. This already helps a little. Now I know a new build with .NET 4.7.2 probably has not started rolling out yet, for example. But knowing in advance when thing are planned to roll out would be even better 😉

**Comment by bryanmacfarlane** [comment]
yeah, we have another issue to track adding a timeline / roadmap document to this repo.  The short answer is we target the middle of the month and we grab latest of that channel (vs2017, etc...) at gen time.  Sometimes we slip.  For example, we're way behind this month because SSDT is still failing blocking image gen but we're working with that team to get unblocked.  That image will carry 4.7.2.

**Comment by lokesh755** [comment]
Hosted VS2017 agents now has .NET 4.7.2

**Comment by pundj** [comment]
.NET 4.7.2 is on the Hosted VS2017 agents now, but when will it get the following?
Microsoft.Net.Component.4.7.2.SDK
Microsoft.Net.Component.4.7.2.TargetingPack
Microsoft.Net.ComponentGroup.4.7.2.DeveloperTools
